### PR TITLE
Value Parameter in Magic Number Rule

### LINKED ISSUE #869: Magic number rule failure
**Diktat version: 0.5.2**

```
[ERROR] 2021-05-08 23:21:20 Internal error has occurred in magic-number. Please make an issue on this bug at https://github.com/cqfn/diKTat/.
java.lang.ClassCastException: class org.jetbrains.kotlin.psi.KtParameter cannot be cast to class org.jetbrains.kotlin.psi.KtProperty (org.jetbrains.kotlin.psi.KtParameter and org.jetbr
ains.kotlin.psi.KtProperty are in unnamed module of loader 'app')
        at org.cqfn.diktat.ruleset.rules.chapter3.MagicNumberRule$checkNumber$isLocalVariable$1.invoke(MagicNumberRule.kt:52)
        at org.cqfn.diktat.ruleset.rules.chapter3.MagicNumberRule$checkNumber$isLocalVariable$1.invoke(MagicNumberRule.kt:29)
        at com.pinterest.ktlint.core.ast.PackageKt.parent(package.kt:159)
        at com.pinterest.ktlint.core.ast.PackageKt.parent$default(package.kt:156)
        at org.cqfn.diktat.ruleset.rules.chapter3.MagicNumberRule.checkNumber(MagicNumberRule.kt:52)
        at org.cqfn.diktat.ruleset.rules.chapter3.MagicNumberRule.logic(MagicNumberRule.kt:41)
        at org.cqfn.diktat.ruleset.rules.DiktatRule.visit(DiktatRule.kt:46)
        at com.pinterest.ktlint.core.KtLint$format$1.invoke(KtLint.kt:271)
        at com.pinterest.ktlint.core.KtLint$format$1.invoke(KtLint.kt:23)
        at com.pinterest.ktlint.core.KtLint$visitor$2$2.invoke(KtLint.kt:212)
        at com.pinterest.ktlint.core.KtLint$visitor$2$2.invoke(KtLint.kt:23)
        at com.pinterest.ktlint.core.ast.PackageKt.visit(package.kt:224)
        at com.pinterest.ktlint.core.ast.PackageKt.visit(package.kt:225)
        at com.pinterest.ktlint.core.ast.PackageKt.visit(package.kt:225)
        at com.pinterest.ktlint.core.ast.PackageKt.visit(package.kt:225)
        at com.pinterest.ktlint.core.ast.PackageKt.visit(package.kt:225)
        at com.pinterest.ktlint.core.ast.PackageKt.visit(package.kt:225)
        at com.pinterest.ktlint.core.KtLint$visitor$2.invoke(KtLint.kt:211)
        at com.pinterest.ktlint.core.KtLint$visitor$2.invoke(KtLint.kt:23)
        at com.pinterest.ktlint.core.KtLint.format(KtLint.kt:264)
        at com.pinterest.ktlint.internal.FileUtilsKt.formatFile(FileUtils.kt:79)
        at com.pinterest.ktlint.KtlintCommandLine.process(Main.kt:331)
        at com.pinterest.ktlint.KtlintCommandLine.access$process(Main.kt:116)
        at com.pinterest.ktlint.KtlintCommandLine$lintFiles$2$1.call(Main.kt:257)
        at com.pinterest.ktlint.KtlintCommandLine$lintFiles$2$1.call(Main.kt:116)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
        at java.base/java.lang.Thread.run(Thread.java:832)
[ERROR] 2021-05-08 23:21:20 As a workaround you can disable these inspections in yml config: [MAGIC_NUMBER]
```

### PR DESCRIPTION
**Pull request by kentr0w**
###  What's done:
 - [x] Added new flag to handle value parameter
 - [x] Added this flag to config files
 - [x] Added tests
 - [ ] Closes #869

### CODE REVIEW

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/utils/AstNodeUtils.kt [hunk 1]
```diff
@@ -351,8 +352,8 @@ fun ASTNode.hasModifier(modifier: IElementType) = this.findChildByType(MODIFIER_
  * Checks whether [this] node of type PROPERTY is `var`
  */
 fun ASTNode.isVarProperty() =
-        this.getChildren(null)
-            .any { it.elementType == ElementType.VAR_KEYWORD }
+        this.elementType == PROPERTY &&
```
Comment (petertrr) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/utils/AstNodeUtils.kt:355]: Why were you calling `isVarProperty` for non-property node in the first place?

Comment (kentr0w) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/utils/AstNodeUtils.kt:355]: Sure. Removed property verification from util

File: diktat-analysis.yml [hunk 2]
```diff
@@ -312,6 +312,8 @@
     ignorePropertyDeclaration: "false"
     # Is ignore local variable
     ignoreLocalVariableDeclaration: "false"
+    # Is ignore value parameter
+    ignoreValueParaeter: "true"
```
[the remark this answers is not in the document]
Comment (kentr0w) [reply diktat-analysis.yml:316]: Fixed

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/MagicNumberRule.kt [hunk 3]
```diff
@@ -49,7 +50,8 @@ class MagicNumberRule(configRules: List<RulesConfig>) : DiktatRule(
         val isHashFunction = node.parent({ it.elementType == FUN && it.isHashFun() }) != null
         val isConstant = node.parent({ it.elementType == PROPERTY && it.isConstant() }) != null
         val isPropertyDeclaration = !isConstant && node.parent({ it.elementType == PROPERTY && !it.isNodeFromCompanionObject() }) != null
-        val isLocalVariable = node.parent({ it.isVarProperty() && (it.psi as KtProperty).isLocal }) != null
+        val isLocalVariable = node.parent({ it.elementType == PROPERTY && it.isVarProperty() && (it.psi as KtProperty).isLocal }) != null
+        val isValueParameter = node.parent({ it.elementType == VALUE_PARAMETER }) != null
```
[the remark this answers is not in the document]
Comment (kentr0w) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/MagicNumberRule.kt:54]: Descussed: decided not to change

**Pull request merged**